nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/filter/NotEqualsFilter.java

```diff
@@ -30,7 +30,7 @@ public NotEqualsFilter(final RecordPathSegment lhs, final RecordPathSegment rhs)
     protected boolean test(final FieldValue fieldValue, final Object rhsValue) {
         final Object lhsValue = fieldValue.getValue();
         if (lhsValue == null) {
-            return rhsValue != null;
+            return false;
```

Comment:
I've seen both ways of handling this case in different platforms, where you're comparing null to null. null == null is fairly straight-forward, but null != null gets a bit weird... some platforms return true, some return false. Originally I had it returning true in such cases. But that means that we had to do some weird stuff in PredicatePath, which caused a lot of other problems when we added functions.... so I changed that to make PredicatePath work a little more cleanly. There's a pretty big blurb of inline comments in there about how we were handling it before vs now.